### PR TITLE
Fix telegram dst default ca dir

### PR DESCRIPTION
**Pull request by lbudai**
PR #2410 breaks the Telegram destination.

This patch adds new option to HTTP destination (`enable-default-ca-dir`).
When defaults are enabled and the user does not set the `ca-dir` option,a default ca-dir is used, provided by the CURL library.

When both options are set (`ca-dir` and `enable-default-ca-dir`), of course, `ca-dir` will be used.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

File: modules/http/http.c [hunk 1]
```diff
@@ -103,7 +103,8 @@ _setup_static_options_in_curl(HTTPDestinationWorker *self)
   if (owner->user_agent)
     curl_easy_setopt(self->curl, CURLOPT_USERAGENT, owner->user_agent);
 
-  curl_easy_setopt(self->curl, CURLOPT_CAPATH, owner->ca_dir);
+  if (owner->ca_dir || owner->enable_default_ca_dir)
```
Comment (MrAnno) [inline modules/http/http.c]: Shouldn't we negate the right side?

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/http/http-grammar.ym [hunk 2]
```diff
@@ -134,6 +135,7 @@ http_tls_options
 
 http_tls_option
     : KW_CA_DIR     '(' string ')'            { http_dd_set_ca_dir(last_driver, $3); free($3); }
+    | KW_ENABLE_DEFAULT_CA_DIR '(' yesno ')'  { http_dd_set_enable_default_ca_dir(last_driver, $3); free($3); }
```
Comment (MrAnno) [inline modules/http/http-grammar.ym]: Oops, I haven't noticed, but this `free()` should be removed.

Comment (lbudai) [reply modules/http/http-grammar.ym]: copy-paste :/ (I've just wanted to close this topic as fast as possible, but maybe too fast...)

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by lbudai** [comment]
@bazsi: I heard that you don't really like the name of the option :) I'm open to replace it to (almost...) anything (`use-default-ca-dir`, whatever).

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by bazsi** [comment]
I was thinking to make it less closely related to ca-dir option, and make
it something like:

use-system-cert-store(yes/no) and that would affect both ca-dir() and
ca-file().

I am not completely satisfied with this either, my point is that it should
relate to the system ca store as a concept and not specific to one part of
it (e.g. ca-dir).

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**